croutonjs/src/js/crouton-map.js

```diff
@@ -0,0 +1,359 @@
+function CroutonMap(config) {
+    var self = this;
+	self.config = {
+		google_api_key: null,		  // Required if using the show_map option.  Be sure to add an HTTP restriction as well.
+		distance_units: 'mi'
+	};
+	Object.assign(self.config, config);
+    self.map = null;
+    self.geocoder = null;
+    self.map_objects = [];
+    self.map_clusters = [];
+    self.oms = null;
+    self.markerClusterer = null;
+	self.handlebarMapOptions = null;
+	self.loadGapi = function(callbackFunctionName) {
+		var tag = document.createElement('script');
+		tag.src = "https://maps.googleapis.com/maps/api/js?key=" + self.config['google_api_key'] + "&callback=" + callbackFunctionName;
+		tag.defer = true;
+		tag.async = true;
+		var firstScriptTag = document.getElementsByTagName('script')[0];
+		firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+	};
+	self.getCurrentLocation = function(callback) {
+		if (navigator.geolocation) {
+			navigator.geolocation.getCurrentPosition(callback, self.errorHandler);
+		} else {
+			$('.geo').removeClass("hide").addClass("show").html('<p>Geolocation is not supported by your browser</p>');
+		}
+	};
+
+	self.addCurrentLocationPin = function(latitude, longitude) {
+		var latlng = new google.maps.LatLng(latitude, longitude);
+		self.map.setCenter(latlng);
+
+		var currentLocationMarker = new google.maps.Marker({
+			"map": self.map,
+			"position": latlng,
+		});
+
+		self.addToMapObjectCollection(currentLocationMarker);
+
+		// TODO: needs to show on click only
```

Comment:
is this still a todo or can we remove this comment

Reply:
No, there really was a todo here.  Thanks for catching it.  The fix is now committed.